### DISCUSSION TITLE
Failure to login with PAM when using unixd

### DISCUSSION THREAD

**Discussion by Water261** [kanidm/kanidm#3112] [Q&A] [answered]
After setting up posix integration as shown in the book, trying to login with a kanidm managed account results with PAM rejecting with the `login incorrect` message.

![image](https://github.com/user-attachments/assets/51fcbffc-1514-4338-a553-03867baef33c)
Potentially suspect log from unixd ^

**Comment by Firstyear** [comment]
That's just saying that something that was cached, no longer exists or no longer has posix attrs, so that's fine.

What do you get with `kanidm-unix auth-test --name username`  ?

**Reply by Firstyear** [reply]
And also `kanidm-unix status`

**Reply by Water261** [reply]
![image](https://github.com/user-attachments/assets/9b1cb69b-a0f2-4316-905f-7e62032d0524)

Running the above command with RUST_LOG=debug:
![image](https://github.com/user-attachments/assets/2c24f380-3e0e-420f-8ae0-9fe0b8a3b863)

status reports working too

**Reply by Firstyear** [reply]
Right, so the fact that kanidm-unix auth-test shows a failure means pam won't work. If you do the auth-test again and check the logs in unixd what do you see? What about the kanidmd server logs at the same time?

**Reply by Water261** [reply]
kanidm-unixd logs: 
![image](https://github.com/user-attachments/assets/3b28738d-6184-4e75-bcba-4b8dc0ab8760)

kanidmd logs:
```
f6852132-a4e4-41c1-9fd3-a375a5f2375b INFO           ┕━ ｉ [info]: search | event_tag_id: 10 | initiator: User( anonymous@idm.water261.au, 00000000-0000-0000-0000-ffffffffffff ) (13ccfc2d-057d-433e-bfb2-72190665c5ca, read only)
f6852132-a4e4-41c1-9fd3-a375a5f2375b INFO           ┝━ validate_client_auth_info_to_ident [ 754µs | 49.92% ]
f6852132-a4e4-41c1-9fd3-a375a5f2375b INFO        ┕━ handle_internalunixusertokenread [ 1.30ms | 36.14% / 86.06% ]
f6852132-a4e4-41c1-9fd3-a375a5f2375b INFO     ┕━ account_id_unix_token [ 1.31ms | 0.71% / 86.77% ]
f6852132-a4e4-41c1-9fd3-a375a5f2375b INFO     request [ 1.51ms | 13.23% / 100.00% ] method: GET | uri: /v1/account/matthew@idm.water261.au/_unix/_token | version: HTTP/1.1
```

**Reply by Firstyear** [reply]
"no allowed groups in configuration"

What is the setting for "pam_allowed_groups" in unixd?

**Reply by Water261** [reply]
I figured it out. I copied the config from 1.4.0-dev rather than 1.3.3

**Reply by Water261** [reply]
Changing it to the 1.3.3 format gives success instead

**Reply by Water261** [reply]
Hm, changing the config fixes `auth-test` but still fails on PAM login

**Reply by Water261** [reply]
> "no allowed groups in configuration"
> 
> What is the setting for "pam_allowed_groups" in unixd?

```toml
pam_allowed_login_groups = ["development-virtual-machines"]
```

The user is a member of the group and after fixing the config version mismatch, I get `account success!` now. However attempting to login through a tty rejects with `Login incorrect`

**Reply by Firstyear** [reply]
So you can ssh in but not login via a tty?

**Reply by Water261** [reply]
Unsure, haven't had a chance to test ssh yet.

**Comment by Water261** [comment, marked as answer]
Figured out as to why it broke, I spelt pam_succeed_if wrong.

**Reply by yaleman** [reply]
That'll do it. It's always the simplest things that are hard to spot. Good to hear it's working!
